client/components/edit/children/SelectableChild.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import { useEditorContext } from "../../../context/EditorContext";
+import ChildList from "./ChildList";
+import AddBox from "@mui/icons-material/AddBox";
+import IndeterminateCheckBox from "@mui/icons-material/IndeterminateCheckBox";
+import { extractFirstMetadataValue } from "../../../util/metadata";
+
+export interface SelectableChildProps {
+    pid: string;
+    selectCallback: (pid: string) => void;
+    initialTitle: string;
+}
+
+export const SelectableChild = ({ pid, selectCallback, initialTitle }: SelectableChildProps): React.ReactElement => {
+    const {
+        state: { objectDetailsStorage },
+    } = useEditorContext();
+    const [expanded, setExpanded] = useState<boolean>(false);
+    const loaded = Object.prototype.hasOwnProperty.call(objectDetailsStorage, pid);
```

Comment:
Is there a way to filter out the null pids before rendering components so you don't have to do this additional check?

Reply:
I don't think so. When we render the list of children, we have PIDs and titles from the Solr index, but these might be out of date. Wiring each SelectableChild up to the objectDetailsStorage ensures that we display the most up-to-date state of those objects (and redraw them if other operations on the page cause changes to Fedora). It feels to me like trying to account for this at a higher level would be more complicated and might lead to larger re-renders than are necessary.